rayon-core/src/registry.rs

```diff
@@ -520,6 +536,19 @@ unsafe fn main_loop(worker: Worker<JobRef>, registry: Arc<Registry>, index: usiz
 
     // Normal termination, do not abort.
     mem::forget(abort_guard);
+
+    // Inform a user callback that we exited a thread.
```

Comment:
(Do we want to add a test for this? Not sure it's needed, given that there are existing tests of panic propagation.)